api/spec/v1.yaml

```diff
@@ -1080,6 +1080,29 @@ paths:
                 $ref: '#/components/schemas/RuntimeAccount'
         <<: *common_error_responses
 
+  /{runtime}/accounts/{address}/evm_nfts:
+    get:
+      summary: |
+        Returns the list of non-fungible token (NFT) instances owned by an account.
+      parameters:
```

Comment:
Would it be possible to add an optional parameter for filtering by token contract address? (According to the current UI design, when browsing the inventory belonging to an account, we are going to display the different collections separately.)

Reply:
how does filtering by token contract address fit into this? will it also need a query to get the list of token contracts that the account owns at least one of?

Reply:
> will it also need a query to get the list of token contracts that the account owns at least one of?

I think we already have that info. 
https://nexus.oasis.io/v1/sapphire/accounts/oasis1qzxpkka97g4pvl9ghc5da2pharwvzth56uymlsgv
returns
``` JSON
{
  "address": "oasis1qzxpkka97g4pvl9ghc5da2pharwvzth56uymlsgv",
  "address_preimage": {
    "address_data": "7QPqnJbsOQl1SCVuQooWPl9STkc=",
    "context": "oasis-runtime-sdk/address: secp256k1eth",
    "context_version": 0
  },
  "balances": [
    {
      "balance": "1126883292480000000000",
      "token_decimals": 18,
      "token_symbol": "ROSE"
    }
  ],
  "evm_balances": [
    {
      "balance": "42",
      "token_contract_addr": "oasis1qpwe6h0hw6s0x4skujhznld065300p83wgxnvu0n",
      "token_contract_addr_eth": "0x998633BDF6eE32A9CcA6c9A247F428596e8e65d8",
      "token_decimals": 0,
      "token_name": "NFTrout",
      "token_symbol": "TROUT",
      "token_type": "ERC721"
    },
    {
      "balance": "23",
      "token_contract_addr": "oasis1qqkcynmpm5tj9thyfq0fg5k2slvdap9teu02n77d",
      "token_contract_addr_eth": "0xFcfed3be2d333F24854cA8d3A351E772272D5842",
      "token_decimals": 0,
      "token_name": "NFTrout Gen 1",
      "token_symbol": "TROUT1",
      "token_type": "ERC721"
    }
  ],
  "stats": {
    "num_txns": 55,
    "total_received": "4000000000000000000000",
    "total_sent": "0"
  }
}
```

Here, the `evm_balances` field lists the tokens, and our balance. If the balance is not 0, we have some of it.

Reply:
> how does filtering by token contract address fit into this?

When looking at tokens, first we list all the collections we have anything from. We can already do that based on the data we get as part of the account info.

Then as a next step, if the user clicks into a collection, then we need to list all the instances the user owns from that collection.

Now, we could do the filtering for collection (based on contract address) locally on the client, but it would be cumbersome, because of pagination. So it would be best if we could query directly the instances that a given account owns, from a given collection.

Reply:
ah ok

Reply:
adding in #539

Reply:
resolving here, let's discuss there